pupil-spa/src/app/services/submission/submission.service.ts

```diff
@@ -21,7 +24,11 @@ export class SubmissionService {
     await this.http.post(`${environment.apiURL}/api/check-started`,
       { checkCode, accessToken },
       requestArgs)
-      .retryWhen(errors => errors.delay(apiErrorDelay).take(apiErrorMaxAttempts))
+      .retryWhen((errors) => {
+        return errors.mergeMap((error) =>
+          (error.status !== 503 && error.status !== 504) ? Observable.throw(error) : Observable.of(error))
```

Comment:
looks good.  Please remove status code filtering, and send each error to the audit log, if any occur.

Reply:
@GuyHarwood - why would you retry in all cases?  Please explain.

Reply:
@jon-shipley because such a narrow range of status codes assumes these are the only situations in which we would want to try again.  Let's say that we miss a required env var on deployment, and the server is throwing 500s.  We correct it, and seconds later the app is up and running.  There should be opportunity to recover from the majority of unanticipated situations, not the few.